File: app/src/main/java/com/sermage/mymoviecollection/data/MovieDao.java
package com.sermage.mymoviecollection.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.sermage.mymoviecollection.pojo.Movie;

import java.util.List;
@Dao
public interface MovieDao {
    @Query("SELECT * FROM Movies")
    LiveData<List<Movie>> getAllMovies();

    @Query("SELECT * FROM favourite_movie")
    LiveData<List<FavouriteMovie>> getAllFavouriteMovies();

    @Query("SELECT * FROM Movies WHERE id==:movieId")
    Movie getMovieById(int movieId);

    @Query("SELECT * FROM favourite_movie WHERE id==:movieId")
    FavouriteMovie getFavouriteMovieById(int movieId);

    @Query("DELETE FROM Movies")
    void deleteAllMovies();

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    void insertMovies(List<Movie> movies);

    @Insert
    void insertFavouriteMovie(FavouriteMovie movie);

    @Delete
    void deleteMovie(Movie movie);

    @Delete
    void deleteFavouriteMovie(FavouriteMovie movie);

}

File: app/src/main/java/com/sermage/mymoviecollection/data/MovieDao.kt
package com.sermage.mymoviecollection.data

import androidx.lifecycle.LiveData
import androidx.room.*
import com.sermage.mymoviecollection.pojo.Movie
import com.sermage.mymoviecollection.pojo.TVShow

@Dao
interface MovieDao {

    @Query("SELECT*FROM favorite_movies")
    fun getFavoriteMovies():LiveData<List<Movie>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertMovie(movie: Movie)

    @Delete
    suspend fun deleteMovie(movie: Movie)

    @Query("SELECT * FROM favorite_movies WHERE id==:movieId")
    suspend fun getMovieById(movieId: Int): Movie

    @Query("SELECT*FROM favorite_tv_shows")
    fun getFavoriteTVShows():LiveData<List<TVShow>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertTVShow(tvShow: TVShow)

    @Delete
    suspend fun deleteTVshow(tvShow: TVShow)

    @Query("SELECT * FROM favorite_tv_shows WHERE id==:tvShowId")
    suspend fun getTVShowById(tvShowId: Int): TVShow



}